storage/badger/operation/computation_result.go

```diff
@@ -0,0 +1,45 @@
+package operation
+
+import (
+	"github.com/dgraph-io/badger/v2"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// InsertComputationResult addes given instance of ComputationResult into local BadgerDB.
+func InsertComputationResultUploadStatus(computationResultID flow.Identifier,
+	wasUploadCompleted bool) func(*badger.Txn) error {
+	return insert(makePrefix(codeComputationResults, computationResultID), wasUploadCompleted)
+}
+
+// RemoveComputationResult removes an instance of ComputationResult with given ID.
+func RemoveComputationResultUploadStatus(
+	computationResultID flow.Identifier) func(*badger.Txn) error {
+	return remove(makePrefix(codeComputationResults, computationResultID))
+}
+
+// GetComputationResult returns stored ComputationResult instance with given ID.
+func GetComputationResultUploadStatus(computationResultID flow.Identifier,
+	wasUploadCompleted *bool) func(*badger.Txn) error {
+	return retrieve(makePrefix(codeComputationResults, computationResultID), wasUploadCompleted)
+}
+
+// GetAllComputationResultIDs returns all IDs of stored ComputationResult instances.
+func GetAllComputationResultIDs(computationResultIDs *[]flow.Identifier) func(*badger.Txn) error {
+	return traverse(makePrefix(codeComputationResults), func() (checkFunc, createFunc, handleFunc) {
+		check := func(key []byte) bool {
+			*computationResultIDs = append(*computationResultIDs, flow.HashToID(key[1:]))
```

Comment:
I think we usually only just check the entry in this function, and collect data in `handle` func

Reply:
i grep our code base and only found that only the value part in the key-value pair is returned in `handle`, and the only place to get key is in `checkFunc`. But let me know if we can indeed get key from `handle`, it is indeed semantically better.

Reply:
ah yes, right!